streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -677,10 +662,10 @@ public synchronized <KIn, VIn, KOut, VOut> Topology addProcessor(final String na
      *
      * @throws TopologyException
      *         if the {@link StoreBuilder#name() state store} was already added, or
-     *         if a processor name is unknown
+     *         if a processor name is unknown or specifies a source or sink
      */
-    public synchronized Topology addStateStore(final StoreBuilder<?> storeBuilder,
-                                               final String... processorNames) {
+    public synchronized <S extends StateStore> Topology addStateStore(final StoreBuilder<S> storeBuilder,
```

Comment:
Decided to all move off `<?>` for the store type... Does not add real value, but for concistency.